cli/lib/nf3.mjs

```diff
@@ -119,6 +113,28 @@ class Nf3 {
   */
   setzkpKeys(keys) {
     this.zkpKeys = keys;
+    this.subscribeToIncomingViewingKeys();
+  }
+
+  /**
+  Setter for the zkp keys by mnemonic, in case it wasn't known at build time and we don't
+  want to use autogenerated ones.
+  @method
+  @param {string} mnemonic - 12 word phrase
+  @param {number} addressIndex - Index used to generate keys combined with mnemonic
+  */
+  async setzkpKeysFromMnemonic(mnemonic, addressIndex) {
+    if (mnemonic !== '') {
+      this.mnemonic.phrase = mnemonic;
+    }
+    this.mnemonic.addressIndex = addressIndex.toString();
+    this.zkpKeys = (
+      await axios.post(`${this.clientBaseUrl}/generate-keys`, {
+        mnemonic: this.mnemonic.phrase,
+        path: `m/44'/60'/0'/${this.mnemonic.addressIndex}`,
+      })
+    ).data;
+    this.subscribeToIncomingViewingKeys();
```

Comment:
the `this.subscribeToIncomingViewingKeys()` Promise needs to be returned, otherwise the code which is awaiting `setzkpKeysFromMnemonic` will run before `subscribeToIncomingViewingKeys()` has finished.

Reply:
Ok, Should i do the same change here? ```  setzkpKeys(keys) {
    this.zkpKeys = keys;
    this.subscribeToIncomingViewingKeys();
  }
```

Reply:
Yes, good spot.